### ISSUE TITLE
Full backup by schedule does not work for a second database with the same name in a different namespace

### ISSUE THREAD

**Issue by Ratio2** [percona/percona-postgresql-operator#693]
### Report

.

### More about the problem

.

### Steps to reproduce

```shell
helm --namespace=test-operator upgrade --install --create-namespace --set=watchAllNamespaces=true --repo=https://percona.github.io/percona-helm-charts pg-operator pg-operator
helm --namespace=test upgrade --install --create-namespace --repo=https://percona.github.io/percona-helm-charts test pg-db
helm --namespace=test2 upgrade --install --create-namespace --repo=https://percona.github.io/percona-helm-charts test pg-db
```

### Versions

1. Kubernetes 1.27
2. Operator 2.3.4
3. Database pg-db 2.3.5

**Comment by Ratio2** [comment]
```2024-03-18T10:12:47.525Z	ERROR	Reconciler error	{"controller": "perconapgcluster", "controllerGroup": "pgv2.percona.com", "controllerKind": "PerconaPGCluster", "PerconaPGCluster": {"name":"test-pg-db","namespace":"test3"}, "namespace": "test3", "name": "test-pg-db", "reconcileID": "5060d1b6-b7d7-42db-ba50-5b8d44d216cc", "error": "get PostgresCluster: PostgresCluster.postgres-operator.crunchydata.com \"test-pg-db\" not found", "errorVerbose": "PostgresCluster.postgres-operator.crunchydata.com \"test-pg-db\" not found\nget PostgresCluster\ngithub.com/percona/percona-postgresql-operator/percona/controller/pgcluster.(*PGClusterReconciler).Reconcile\n\t/go/src/github.com/percona/percona-postgresql-operator/percona/controller/pgcluster/controller.go:245\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.3/pkg/internal/controller/controller.go:119\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.3/pkg/internal/controller/controller.go:316\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.3/pkg/internal/controller/controller.go:266\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.3/pkg/internal/controller/controller.go:227\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1650"}
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.3/pkg/internal/controller/controller.go:266
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.3/pkg/internal/controller/controller.go:227
```

**Comment by spron-in** [comment]
Hello @Ratio2 - I created this issue for us to track it: https://perconadev.atlassian.net/browse/K8SPG-531

Thank you for reporting it.

**Comment by hors** [comment]
@Ratio2 the issue was fixed in PGO 2.4.0. JFYI PGO 2.5.0 was released two days ago. You can use the latest version.
